huaweicloud/services/cce/resource_huaweicloud_cce_partition_v3.go

```diff
@@ -0,0 +1,220 @@
+package cce
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/chnsz/golangsdk/openstack/cce/v3/partitions"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/logp"
+)
+
+func ResourceCCEPartitionV3() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceCCEPartitionV3Create,
+		ReadContext:   resourceCCEPartitionV3Read,
+		UpdateContext: resourceCCEPartitionV3Update,
+		DeleteContext: resourceCCEPartitionV3Delete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceCCEPartitionV3Import,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"cluster_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"availability_zone": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"category": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"partition_subnet_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"container_subnet_ids": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+		},
+	}
+}
+
+func resourceCCEPartitionSubnetID(d *schema.ResourceData) partitions.HostNetwork {
+	return partitions.HostNetwork{
+		SubnetID: d.Get("partition_subnet_id").(string),
+	}
+}
+
+func resourceCCEContainerSubnetIDs(d *schema.ResourceData) []partitions.ContainerNetwork {
+	networkRaw := d.Get("container_subnet_ids").(*schema.Set)
+	containerNetwork := make([]partitions.ContainerNetwork, networkRaw.Len())
+	for i, raw := range networkRaw.List() {
+		containerNetwork[i] = partitions.ContainerNetwork{
+			SubnetID: raw.(string),
+		}
+	}
+	return containerNetwork
+}
+
+func resourceCCEPartitionV3Create(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	cceClient, err := config.CceV3Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating HuaweiCloud CCE Partition client: %s", err)
```

Comment:
don't use fmtp and logp, remove 'HuaweiCloud' from the error and log messages
```suggestion
		return diag.Errorf("Error creating HuaweiCloud CCE Partition client: %s", err)
```